src/Microsoft.ML.Data/Evaluators/RankingEvaluator.cs

```diff
@@ -26,21 +26,24 @@
 
 namespace Microsoft.ML.Data
 {
-    [BestFriend]
-    internal sealed class RankingEvaluator : EvaluatorBase<RankingEvaluator.Aggregator>
+    /// <summary>
+    ///  Options to control the output of the RankingEvaluator
+    /// </summary>
+    public sealed class RankingEvaluatorOptions
     {
-        public sealed class Arguments
-        {
-            [Argument(ArgumentType.AtMostOnce, HelpText = "Maximum truncation level for computing (N)DCG", ShortName = "t")]
-            public int DcgTruncationLevel = 3;
+        [Argument(ArgumentType.AtMostOnce, HelpText = "Maximum truncation level for computing (N)DCG", ShortName = "t")]
+        public int DcgTruncationLevel = 3;
 
-            [Argument(ArgumentType.AtMostOnce, HelpText = "Label relevance gains", ShortName = "gains")]
-            public string LabelGains = "0,3,7,15,31";
+        [Argument(ArgumentType.AtMostOnce, HelpText = "Label relevance gains", ShortName = "gains")]
+        public string LabelGains = "0,3,7,15,31";
 
-            [Argument(ArgumentType.AtMostOnce, HelpText = "Generate per-group (N)DCG", ShortName = "ogs")]
-            public bool OutputGroupSummary;
-        }
+        [Argument(ArgumentType.AtMostOnce, HelpText = "Generate per-group (N)DCG", ShortName = "ogs")]
+        public bool OutputGroupSummary;
```

Comment:
From what I can tell by looking through the code, `OutputGroupSummary` has no affect on the output metrics that are returned on the public API.

I think we should either
1. Remove the public option
2. Expose the extra metrics (the group summary) this option controls

A hybrid approach would be to do number (1) now, and log a bug to do (2) later.

Reply:
By "Remove the public option" do you mean to make the OutputGroupSummary member private to the class? Won't that break the usage in the RankingEvaluator class? 
Can we keep it public for now and log a bug to do (2) later?


---
In reply to: [317249875](https://github.com/dotnet/machinelearning/pull/4081#discussion_r317249875) [](ancestors = 317249875)

Reply:
I wouldn’t recommend adding a public API that doesn’t do anything. My preference would be to add the public API when it actually accomplishes something for the user.

Reply:
Is it acceptable to mark the field internal and leave the rest of the code intact? Or should we remove the variable entirely from the definition of RankingEvaluatorOptions?

---
In reply to: [323032537](https://github.com/dotnet/machinelearning/pull/4081#discussion_r323032537) [](ancestors = 323032537)

Reply:
Marking it internal is totally acceptable.